pkgs/applications/editors/neovim/wrapper.nix

```diff
@@ -40,22 +47,59 @@ let
     # lua code to put into the generated init.lua file
     , luaRcContent ? ""
     # entry to load in packpath
-    , packpathDirs
+    , packpathDirs ? null # not used anymore
+    , plugins ? []
     , ...
-  }:
+  }@attrs:
   assert withPython2 -> throw "Python2 support has been removed from the neovim wrapper, please remove withPython2 and python2Env.";
 
+  # assert packpathDirs -> throw "Packpathdirs removed";
+
   stdenv.mkDerivation (finalAttrs:
   let
+    pluginsNormalized = neovimUtils.normalizePlugins plugins;
+
+    myVimPackage = neovimUtils.normalizedPluginsToVimPackage pluginsNormalized;
+
+    rubyEnv = bundlerEnv {
+      name = "neovim-ruby-env";
+      gemdir = ./ruby_provider;
+      postBuild = ''
+        ln -sf ${ruby}/bin/* $out/bin
+      '';
+    };
+
+    pluginRC = lib.foldl (acc: p: if p.config != null then acc ++ [p.config] else acc) []  pluginsNormalized;
+
+    manifestRc = vimUtils.vimrcContent { customRC = ""; };
+    # we call vimrcContent without 'packages' to avoid the init.vim generation
+    neovimRcContent' = vimUtils.vimrcContent {
+      beforePlugins = "";
+      customRC = lib.concatStringsSep "\n" (pluginRC ++ [neovimRcContent]);
+      packages = null;
+    };
 
     finalPackdir = neovimUtils.packDir packpathDirs;
 
     rcContent = ''
       ${luaRcContent}
-    '' + lib.optionalString (!isNull neovimRcContent) ''
-      vim.cmd.source "${writeText "init.vim" neovimRcContent}"
+    '' + lib.optionalString (!isNull neovimRcContent') ''
```

Comment:
IIRC `isNull` is deprecated, if the line is being touched anyway we can migrate to `!= null`?

Reply:
didn't know about that: do you have a reference ? isNull is a builtins so I am surprised it gets removed. But yeah np problem for changing it to `!= null`

Reply:
There's a warning on [Noogle](https://noogle.dev/f/builtins/isNull) and [nix.dev](https://nix.dev/manual/nix/2.18/language/builtins#builtins-isNull). Not a big deal but better to avoid usage if possible 🙂